test/e2e/mutation_test.go

```diff
@@ -38,7 +39,28 @@ func mutationTest(ctx context.Context, c client.Client, invConfig InventoryConfi
 
 	inv := invConfig.InvWrapperFunc(invConfig.InventoryFactoryFunc(inventoryName, namespaceName, "test"))
 
-	podAObj := withNamespace(manifestToUnstructured(podA), namespaceName)
+	podAObj := withMutation(withNamespace(manifestToUnstructured(podA), namespaceName), mutation.ApplyTimeMutation{
```

Comment:
This test is a little too dependent on the impl. It might be better to use go-template to template the input yaml.